tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -634,3 +657,64 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
             anchor.realize()
 
             assert anchor.fullyRealized
+
+    describe 'when scrolling to a virtual anchor', ->
+
+      it 'scrolls right next to the wanted page (to get it rendered)', ->
+        am = createAnchoringManagerAndLazyDocument()
+        ann = createTestAnnotationForPages "a1", [10]
+        anchor = am.createAnchor(ann, ann.target[0]).result
+        am.document.currentIndex = 5  # We start from page 5
+        am.document.setPageIndex = sinon.spy (index) ->
+          am.document.currentIndex = index
+          if index is 9
+            renderPage am.document, 9
+            renderPage am.document, 10
+
+        # Now we trigger the actual action
+        anchor.scrollIntoView().then ->
+          assert.calledWith am.document.setPageIndex, 9
+
+      it 'gets the wanted page rendered', ->
+        am = createAnchoringManagerAndLazyDocument()
+        ann = createTestAnnotationForPages "a1", [10]
+        anchor = am.createAnchor(ann, ann.target[0]).result
+        am.document.currentIndex = 5  # We start from page 5
+        am.document.setPageIndex = sinon.spy (index) ->
+          am.document.currentIndex = index
+          if index is 9
+            renderPage am.document, 9
+            renderPage am.document, 10
+
+        # Now we trigger the actual action
+        anchor.scrollIntoView().then ->
+          assert am.document.isPageMapped 10
+
+      it 'scrolls to the wanted page, eventually', ->
+        am = createAnchoringManagerAndLazyDocument()
+        ann = createTestAnnotationForPages "a1", [10]
+        anchor = am.createAnchor(ann, ann.target[0]).result
+        am.document.currentIndex = 5  # We start from page 5
+        am.document.setPageIndex = sinon.spy (index) ->
+          am.document.currentIndex = index
+          if index is 9
+            renderPage am.document, 9
+            renderPage am.document, 10
```

Comment:
> This looks like it's testing the wiring between the `setPageIndex` and the `renderPage` helper. 

Ummm... no, not really.

> These tests should only be validating the behaviour of the `scrollIntoView` method and that they call appropriate methods on dependencies.

That's exactly what we are doing; the challenge is that they can only perform all the required tasks if the other pieces cooperate. That's why we are mocking them.

Here is a detailed description of the process.

Our players are:
 * `Platform`: The piece of code responsible for rendering the document. Outside of our control. Example: PDF.js
 * `Document Access Strategy` (`docAccess` from now on): The piece of code responsible for interfacing with the specific platform. Implements the d-t-m interface. Examples: d-t-m, pdf-text-mapper
 * `Anchoring Manager`: this is the piece of code we are testing. Responsible for ... surprise: managing anchors.
  * `Application`: Gives high-level commands to the `Anchoring Manager`

Process (of scrolling to a virtual anchor, a.k.a. two-phase scrolling):
 * `Application` calls an Anchor's `scrollIntoView()` method, which is part of the `Anchoring Manager`.
 * The `scrollIntoView()` method recognizes that the wanted page is not rendered yet, so we need two-phase scrolling.
 * The `Anchoring Manager` instructs the `docAccess` to approach the wanted page (typically goes to the immediate neighboring page) - this is phase 1 of the scrolling
 * The `docAccess` requsts the `platform` to go to the specified page (using the platform-specific method to do so)
 * The `platform` goes to the wanted page, which is the neighbor of what we actually want
 * The `platform` takes a guess on the user's next move, and per-emtively renders the pages adjacent to the one currently displayed - including the page that we actually wanted.
 * The `docAccess`, which is monitoring the `platform` using whatever platform-spefic methods are available, recognizes that a new page has been rendered, and notifies the `Anchoring Manager` about this development.
 * In turn, the `Anchoring Manager` will realize all the anchors intersecting the given page, or in other words, it will create the highlights for it.
 * After the new highlights are created for the anchor that we want to scroll to, the `Anchoring Manager` remembers that it has a pending scroll on this anchor, which can be fully executed now, and so calls the `scrollIntoView()` method on the newly created highlight.
 * Then the highlight (which consists of actual DOM elements) will instruct the browser to scroll him into view (currently using the jquery scorllintoview plutin.)
 * The promise returned by the Anchor's `scrollIntoView()` method is resolved.

This is what happens.

In this test case, we are trying to test the `Anchoring Manager`'s behavior in this process.
To do so, we must play the part of the `Application`, and simulate/mock/stub the `Platform`, the `docAccess` and the `Highlight`.

And that's what we try to do.
 * `DummyDocumentAccess` is our fake `docAccess`.
 * The (manually triggered) `renderPage()`, `unrenderPage()` methods take on the role of the `platform`.

We only provide the parts that are required to fully exercise the code that must be tested in the `Anchoring Manager`.

Please advise me if you see any possibility to simplify the test, given these requirements.